src/__tests__/createSchemaForm.test.tsx

```diff
@@ -1847,4 +1861,466 @@ describe("createSchemaForm", () => {
+      const mockOnSubmit = jest.fn();
+      debugger;
+      function ComplexField({}: { complexProp1: boolean }) {
+        return (
+          <div>
+            <div>
+              Number and boolean in a row
+              <FormFragment
+                schema={objectSchema
+                  .pick({ bool: true })
+                  .merge(objectSchema.pick({ num: true }))}
+                props={{ num: { suffix: "%" } }}
+              />
+            </div>
+            <div>
+              String fields in a row
+              <FormFragment schema={objectSchema.pick({ str: true })} />
+            </div>
+          </div>
+        );
+      }
+
+      const objectSchema = z.object({
+        num: z.number(),
+        str: z.string(),
+        bool: z.boolean(),
+      });
+
+      const mapping = [
+        [z.string(), TextField],
+        [z.number(), NumberField],
+        [z.boolean(), BooleanField],
+        [objectSchema, ComplexField],
+      ] as const;
+
+      const [Form, FormFragment] = createTsFormAndFragment(mapping);
+
+      const schema = z.object({
+        nestedField: objectSchema,
+      });
+      const defaultValues = {
+        nestedField: { num: 4, str: "this", bool: true },
+      };
+      const form = (
+        <Form
+          schema={schema}
+          onSubmit={mockOnSubmit}
+          defaultValues={defaultValues}
+          props={{
+            nestedField: { complexProp1: true },
+          }}
+          renderAfter={() => <button type="submit">submit</button>}
+        />
+      );
+      const { rerender } = render(form);
+      const button = screen.getByText("submit");
+      await userEvent.click(button);
+      // this rerender is currently needed because setError seemingly doesn't rerender the component using useController
+      rerender(form);
+      const textNodes = screen.queryByTestId(defaultTextInputTestId);
+      expect(textNodes).toBeInTheDocument();
+      expect(textNodes).toHaveDisplayValue("this");
+      const numberNodes = screen.queryByTestId(defaultNumberInputTestId);
+      expect(numberNodes).toBeInTheDocument();
+      expect(numberNodes).toHaveDisplayValue("4");
+      const booleanNodes = screen.queryByTestId(defaultBooleanInputTestId);
+      expect(booleanNodes).toBeInTheDocument();
+      expect(booleanNodes).toBeChecked();
+      screen
+        .queryAllByTestId(errorMessageTestId)
+        .forEach((node) => expect(node).toBeEmptyDOMElement());
+      expect(mockOnSubmit).toHaveBeenCalledWith(defaultValues);
+    });
+    it("should render recursive object schemas", async () => {
+      const mockOnSubmit = jest.fn();
+      function RecursiveObjectField({
+        name,
+      }: {
+        complexProp1?: boolean;
+        name: string;
+      }) {
+        const namePrefix = useMaybeFieldName();
+        const {
+          field: { value },
+        } = useTsController<z.infer<RecursiveObjectSchema>>({
+          name: [namePrefix, name].join(""),
```

Comment:
here's where i'm making use of the name override to all the component to be reused on line 2229 below